dhcpv6/ztp/parse_vendor_options.go

```diff
@@ -0,0 +1,72 @@
+package ztpv6
+
+import (
+	"errors"
+	"log"
+	"strings"
+
+	"github.com/insomniacslk/dhcp/dhcpv6"
+)
+
+var (
+	errVendorOptionMalformed = errors.New("malformed vendor option")
+)
+
+// VendData contains fields extracted from Option 17 data
+type VendData struct {
+	VendorName, Model, Serial string
+}
+
+// VendorData will try to parse dhcp6 Vendor Specific Information options data
+// looking for more specific vendor data (like model, serial number, etc).
+// If the options are missing we will just return an empty struct
+func VendorData(packet dhcpv6.DHCPv6) (VendData, error) {
```

Comment:
For consistency with the ztpv4 module lets call this `ParseVendorData`. Do we need separate functions for `VendorData` and `parseV6VendorSpecificInfo`? Maybe we can combine them.

Reply:
How about returning a reference to the `VendData` to avoid copying the whole struct?